### PR TITLE
[Qwen3-TTS][XPU] Auto-select the intel_xpu attention backend on XPU

### PR DESCRIPTION
**Pull request by YangKai0616**
## Motivation

Qwen3-TTS falls through to the generic triton attention backend on XPU currently. This PR declares the auto-selection for `Qwen3TTSForConditionalGeneration` through the upstream `@register_model_override registry`.

## Accuracy Test

SeedTTS EN full set (1088 samples, voice cloning, concurrency 16, non-streaming), `Qwen3-TTS-12Hz-1.7B-Base` on 1× Intel Arc Pro B60 with `intel_xpu`:

Metric | This run | Reference (0.6B / H200)
-- | -- | --
WER corpus | 1.005% | 1.07%
WER median / p95 | 0.00% / 8.33% | — / 9.09%
Runaway (>50% WER) | 0 / 1088 | 2 / 1088
Completed / failed | 1088 / 0 | 1088 / 0
Speaker similarity | 71.33% | —

## Benchmark

Decode attention micro-benchmark on B60 (16 q-heads / 8 kv-heads / head_dim 128 / bf16):

bs × seq | intel_xpu | triton | speedup
-- | -- | -- | --
1 × 128 | 0.084 ms | 0.156 ms | 1.85×
4 × 1024 | 0.089 ms | 0.226 ms | 2.54×
16 × 512 | 0.113 ms | 0.394 ms | 3.49×
16 × 2048 | 0.345 ms | 1.116 ms | 3.23×

## Checklist

- [√] Format your code according with pre-commit.
- [ ] Add unit tests.
- [ ] Update documentation / docstrings / example tutorials as needed.
- [√] Provide throughput / latency benchmark results and accuracy evaluation results as needed.
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.

## CI

CI runs on self-hosted GPU runners and requires a maintainer to add the
`run-ci` label. Once labeled, every subsequent push re-triggers CI as
long as the label remains. Use `/tag-and-rerun-ci higgs` or
`/tag-and-rerun-ci moss` or `/tag-and-rerun-ci qwen3-tts` to select a TTS CI
model, and
`/tag-and-rerun-ci fun-asr`, `/tag-and-rerun-ci qwen3-asr` or
`/tag-and-rerun-ci whisper-asr` to select an ASR CI model. One selector from
each family can be combined, for example `/tag-and-rerun-ci moss fun-asr`.
Draft PRs are skipped even if labeled.

### CODE REVIEW

**Comment by YangKai0616** [comment]
Hi @jianan-gu , this PR is xpu specific, pls help review, thx!

File: sglang_omni/models/qwen3_tts/stages.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 from typing import Any
 
 import torch
+from sglang.srt.arg_groups.overrides import register_model_override
```
[the remark this answers is not in the document]
Comment (YangKai0616) [reply sglang_omni/models/qwen3_tts/stages.py:12]: The current repo already pins `sglang==0.5.16`, and the `v0.5.16` tag exists `register_model_override`. Therefore, no changes are needed.

File: sglang_omni/models/qwen3_tts/stages.py [hunk 2]
```diff
@@ -92,6 +93,18 @@ def _patched_init(self, *args, **kwargs):
...
+@register_model_override("Qwen3TTSForConditionalGeneration")
+def _qwen3_tts_overrides(server_args: Any, hf_config: Any) -> dict[str, Any]:
+    """Auto-select the intel_xpu attention backend"""
+    del hf_config
+    if server_args.device != "xpu":
+        return {}
+    if not server_args.is_attention_backend_not_set():
+        return {}
+    logger.warning("Use intel_xpu as attention backend on xpu for Qwen3-TTS model")
+    return {"attention_backend": "intel_xpu"}
```
[the remark this answers is not in the document]
Comment (YangKai0616) [reply sglang_omni/models/qwen3_tts/stages.py:105]: Done.

File: sglang_omni/models/qwen3_tts/stages.py [hunk 3]
```diff
@@ -92,6 +93,18 @@ def _patched_init(self, *args, **kwargs):
...
 
+@register_model_override("Qwen3TTSForConditionalGeneration")
+def _qwen3_tts_overrides(server_args: Any, hf_config: Any) -> dict[str, Any]:
+    """Auto-select the intel_xpu attention backend"""
+    del hf_config
+    if server_args.device != "xpu":
+        return {}
+    if not server_args.is_attention_backend_not_set():
+        return {}
+    logger.warning("Use intel_xpu as attention backend on xpu for Qwen3-TTS model")
```
[the remark this answers is not in the document]
Comment (YangKai0616) [reply sglang_omni/models/qwen3_tts/stages.py:104]: The wording aligns with upstream SGLang’s [`_llama4_overrides`](https://github.com/sgl-project/sglang/blob/ffc431cd4c284e3204413b66f128a588c715b54d/python/sglang/srt/arg_groups/overrides.py#L1162) registry. Changing it to `“Using”` would break consistency with similar upstream providers.

File: sglang_omni/models/qwen3_tts/stages.py [hunk 4]
```diff
@@ -92,6 +93,18 @@ def _patched_init(self, *args, **kwargs):
...
 
 
+@register_model_override("Qwen3TTSForConditionalGeneration")
+def _qwen3_tts_overrides(server_args: Any, hf_config: Any) -> dict[str, Any]:
+    """Auto-select the intel_xpu attention backend"""
+    del hf_config
+    if server_args.device != "xpu":
+        return {}
+    if not server_args.is_attention_backend_not_set():
+        return {}
```
[the remark this answers is not in the document]
Comment (YangKai0616) [reply sglang_omni/models/qwen3_tts/stages.py:103]: `ServerArgs.__post_init__` strips the device index (`self.device.split(":")[0]`) before providers run, and `is_attention_backend_not_set` is the upstream `ServerArgs` API used by 9 providers in the same `overrides.py` — it also covers `prefill/decode_attention_backend`, which a manual is None check would miss. No change needed.

File: tests/unit_test/qwen3_tts/test_server_arg_overrides.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    )
+
+
+@pytest.mark.parametrize(
+    ("device", "attention_backend", "expected"),
+    [
+        ("xpu", None, {"attention_backend": "intel_xpu"}),
+        ("xpu", "triton", {}),
+        ("cuda", None, {}),
+    ],
```
[the remark this answers is not in the document]
Comment (YangKai0616) [reply tests/unit_test/qwen3_tts/test_server_arg_overrides.py:33]: The device is already index-free when the provider runs, so an `xpu:0` case would assert on an input that cannot occur. No change needed.

File: sglang_omni/models/qwen3_tts/stages.py [hunk 6]
```diff
@@ -11,6 +11,8 @@
 from typing import Any
 
 import torch
+from sglang.srt.arg_groups.model_override_base import is_attention_backend_not_set
```
[the remark this answers is not in the document]
Comment (YangKai0616) [reply sglang_omni/models/qwen3_tts/stages.py]: Updated. In the pinned SGLang 0.5.19, `ServerArgs.is_attention_backend_not_set` was moved to a standalone function and is exported under `sglang.srt.arg_groups.overrides`. I've updated the import to use the public export.

File: sglang_omni/models/qwen3_tts/stages.py [hunk 7]
```diff
@@ -143,6 +147,18 @@ def _patched_init(self, *args, **kwargs):
         pass
 
 
+@register_model_override("Qwen3TTSForConditionalGeneration")
+def _qwen3_tts_overrides(server_args: Any, hf_config: Any) -> dict[str, Any]:
```
Comment (jianan-gu) [inline sglang_omni/models/qwen3_tts/stages.py:151]: Overall changes LGTM. while can we have a further check that if we can make this overrides per device dispatching?
